### DISCUSSION TITLE
[TOOL] AzerothCore opensource launcher

### DISCUSSION THREAD

**Discussion by Helias** [azerothcore/azerothcore-wotlk#3908] [Tools]
What about to make a custom opensource "AzerothCore launcher"? Anyone knows some custom launcher that we can fork and improve? May a cross-platform one?

This could help lot of servers that uses custom contents, and we know that after a lot of years of "WOTLK" a lot of servers are trying to add custom contents to improve the play of WOTLK, like I do with the modules which add entire battlegrounds to the game.

P.S.: I don't know if [TOOL] is the right section.

**Comment by pangolp** [comment]
It's an excellent idea, even though I don't know any launchers. I'm always interested in the subject, mainly if you want to add custom content or improve functionality.

**Comment by dunjeon** [comment]
What kind of "features?" I wrote a launcher for my Solo repack, but it is meant to be as simple as possible to use and was designed specifically for my repack data structure . 

We could OpenSource one in .net or Python with a proper development roadmap.

**Comment by Helias** [comment]
.NET is not cross-platform so I think that python is better in this case.
Anyway we can list several features of this project and make a new repo under azerothcore organization to start the project.
Personally I have not time in this period so I can't start it, but what about to make a new launcher in ElectronJS instead of Python?

**Comment by dunjeon** [comment]
The language isn't a barrier of entry, honestly.  I'm open to anything.

**Comment by pangolp** [comment]
Honestly I have no problem with that, python or javascript is fine with me. Either one.

**Comment by Yehonal** [comment]
In 2007 I created a launcher in Pascal Delphi https://github.com/Drassil/wowlauncher-delphi it's very old but it's open source and could be interesting for someone :)

Recently (2 years ago) I've created a Launcher with another guy, written in javascript with Electron, that is capable to download custom patches and do other amazing stuffs. https://app.drassil.net/site/en/download-apps
If you like it I can try to create an open source version, but without contributors it's useless

**Comment by pangolp** [comment]
Doing a little research, "generate a launcher", is no big deal and only works on Windows. But it allows you to delete the cache folder, access the server's website, its head, run the game and change the reamlist.

<a href="https://subefotos.com/ver/?7c653f9611293e485a084ffdbd6f70f5o.png" target="_blank"><img src="http://thumbs.subefotos.com/7c653f9611293e485a084ffdbd6f70f5o.jpg" /></a>

**Comment by BarbzYHOOL** [comment]
electronJs or nw.js (https://nwjs.io/) are crossplatform and with the added benefit of being easy to style (i suppose, as it's like web dev, using html, js, css)

Python is interesting too but no idea how to make a good design with it, so it might be better to stick to the JS versions

And yes, a launcher's main feature is to send custom patches to the user even though there is a way to do it directly with the wowclient (but not implemented on AC yet)

**Comment by solidmaxtor** [comment]
Hey this is by launcher (only launcher game) or for downloader launcher (for download versions/patches and run game) ??

![image](https://user-images.githubusercontent.com/50475454/72193710-a573be80-33d8-11ea-8b4b-0653c1e1a36d.png)

**Reply by icaca** [reply]
hello, where can I download it

**Comment by pangolp** [comment]
@solidmaxtor It just serves to remove the cache, a couple of links to the website and stuff and run the game. Also change the reamlist, but it would miss the important part of downloading the game and updating if there was any patch. I think the best alternative would be to do it at https://electronjs.org/

**Comment by manstfu** [comment]
https://github.com/araxiaonline/wow-client-patcher/tree/main Seems a good starting point
